### ISSUE TITLE
Uploaded SurrealML Model Did Not Give Expected Results

### ISSUE THREAD

**Issue by ReginaPulsifi** [surrealdb/surrealml#52]
**Describe the bug**
I did not get the expected results with the uploaded surrealml model.

**To Reproduce**

1. Created a multiouput linear regression with sklearn
```
import sklearn
import skl2onnx
import numpy as np
from sklearn.linear_model import LinearRegression
from surrealml import SurMlFile, Engine

n_samples = 100
n_features = 3
n_outputs = 3
X = np.array(np.random.rand(n_samples, n_features), dtype=np.float32)
Y = np.array(np.random.rand(n_samples, n_outputs), dtype=np.float32)

model = LinearRegression()
model.fit(X,Y)
```

2. Stored the model as SurrealML File
```
file = SurMlFile(
	model=model, 
	name="testprediction", 
	inputs=X[:1], 
	engine=Engine.SKLEARN
)

file.add_version(version="0.0.1")
file.save(path="./lineartestprediction.surml")

new_file = SurMlFile.load(path="./lineartestprediction.surml", engine=Engine.SKLEARN)
```

3. This is the expected output I get using raw compute in python
![image](https://github.com/surrealdb/surrealml/assets/97146067/a98b05e9-20a3-45a8-817b-e7eb7fd01046)

4. However when I uploaded the model to surrealdb and query it, this is what I get. It only returns the first output, but I expected three
![image](https://github.com/surrealdb/surrealml/assets/97146067/e69165aa-a7c4-4d70-a8fc-fa996dd40d74)

**Expected behavior**
It should return me all the three outputs. [0.5089617371559143, 0.5183728933334351, 0.5182746648788452]

**Environment**
```
Platform: Desktop
OS: Darwin
Architecture: aarch64
WebView: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko)
Version: 2.0.6
Flags: featureFlags: false, models_view: true, apidocs_view: true, themes: false, newsfeed: true
```

**Comment by catethos** [comment]
since in this line: https://github.com/surrealdb/surrealml/blob/9e0b5f6c8c6a0729646104a31721748a1be5b641/modules/core/src/execution/compute.rs#L94, the first layer of the model output is taken

maybe we shouldn't take the first element of the output here:
https://github.com/surrealdb/surrealdb/blob/bf702b0d67c63884dfa7fa567482858ce87315a5/core/src/sql/model.rs#L211

**Comment by catethos** [comment]
duplicated issue here: https://github.com/surrealdb/surrealdb/issues/4098#issuecomment-2133691057

**Comment by catethos** [comment]
hi just checking if this fix will be in the 2.0 release?

**Comment by maxwellflitton** [comment]
@catethos hey sorry for the late reply, it will not be fixed on the 2.0 release but I will be pushing a fix soon. What is holding up the patch is how it will play with the SQL statements. The current form just returns a single number which directly feeds into other SQL statements and columns etc. After looking at it, I am going to add a new function which is `compute_raw_vector` and this will enable users to directly state that they want a vector to be returned. In terms of the ML project in general, even though this repo looks a little stale, a lot of happening behind the scenes. I have been collaborating with Kings College London on async Rust interfaces with GPUs so we can send multiple ML tasks to a GPU. I appreciate your patience on this.

**Comment by maxwellflitton** [comment]
@catethos and @ReginaPulsifi pull request for returning a vector of results is now passing tests for the main database:

https://github.com/surrealdb/surrealdb/pull/4523
